projects/cloud_ml/scripts/scoring.py

```diff
@@ -0,0 +1,100 @@
+import sys
+sys.path.append("/home/yakelyn/explore/yakelyn/utils/")
+import mycolors
+import os
+import fv3viz
+import xarray as xr
+import intake
+import fsspec
+from dask.diagnostics import ProgressBar
+from vcm.catalog import catalog as CATALOG
+from vcm.fv3.metadata import standardize_fv3_diagnostics
+from vcm.safe import get_variables
+from vcm import convert_timestamps
+import cftime
+import numpy as np
+from matplotlib.gridspec import GridSpec
+from matplotlib.colors import LogNorm
+import matplotlib.pyplot as plt
+from cartopy import crs as ccrs
+from sklearn.metrics import mean_squared_error
+from math import sqrt
+import matplotlib.dates as mdates
+import matplotlib.ticker as ticker
+import vcm
+
+def plot_global_map_biases(
+    ds_a,
+    ds_b, 
+    GRID,
+    var_name,
+    cmap,                    
+    units,
+    output_dir,
+    plot_filename,
+    title,
+    title_a,
+    title_b):
+
+    "Plot a 3 row figure. Global maps for ds_a, ds_b, bias = ds_a - ds_b "
+    vmin = np.nanmin(ds_a).round()
+    vmax = np.nanmax(ds_a).round()
+
+    fig = plt.figure(figsize = (4, 6)) 
+    gs = GridSpec(3,1)
+    gs.update(wspace=.35, hspace=0.2)
+
+    ax = fig.add_subplot(gs[0,0],projection=ccrs.Robinson())
+    fv3viz.plot_cube(xr.merge([ds_a, GRID]), var_name, cbar_label= units, ax = ax, vmin = vmin, vmax = vmax ,cmap = cmap)
+    ax.set_title( title +'\n' + title_a, fontsize = 7)
+
+
+    ax = fig.add_subplot(gs[1,0],projection=ccrs.Robinson())
+    fv3viz.plot_cube(xr.merge([ds_b, GRID]), var_name, cbar_label= units, ax = ax, vmin = vmin, vmax = vmax,cmap = cmap)
+    ax.set_title(title_b, fontsize = 7)
+
+
+    ax = fig.add_subplot(gs[2,0],projection=ccrs.Robinson())
+    fv3viz.plot_cube(xr.merge([ds_a - ds_b , GRID]), var_name, cbar_label= units, ax = ax, cmap = mycolors.blues2reds())
+    ax.set_title( title_a + ' - ' + title_b +' bias',fontsize = 7)
+
+    plt.savefig(os.path.join(output_dir, plot_filename), dpi=200, bbox_inches='tight')
+    plt.close('all')
+    return fig
+
+def find_common_times(ds1, ds2):
+    return list(set(ds1.time.values).intersection(set(ds2.time.values)))
+    
+def global_average(x):
+    return x.mean(['x', 'y','tile'])
+
+def sse(truth,prediction):
+    ## returns the sum of squares of the residual errors
+    return (truth-prediction)**2
+
+def ss(truth, prediction):
+    ## returns the total sum of the errors relative its global mean
+    return (truth - global_average(prediction))**2
+
+def r2_score(truth, prediction):
```

Comment:
I'm a little confused by this now ... `ss = global_average(ss(truth, truth))` is the same as the variance of the truth, so isn't this still plain old R**2?

Reply:
Also, avoid duplicating the function name with a variable name (`ss`).

Reply:
> ss = global_average(ss(truth, truth)) is the same as the variance of the truth
Let E be the average. This formula is E[x^2]. The variance is E[(x-E[x])^2]. They are only the same of the mean is zero.

Reply:
Ah yes. I was confused. The indirection of the ss and sse functions confused me. @YakelynRJ I think it might be clearer to remove those functions and copy paste their contents here.

But yes its good we are having this discussion. As i have mentioned before, I think making the denominator be variance often adds a layer of ambiguity about what this score means. For example the r2_integrated_score uses horizontal only average to compute the mean for the denominator and then a (x,y,z) average for the deviations. It's a pretty good demonstration of the pitfalls of using R^2.

 I would advocate for `1 - E[(x-y)^2]/E[x^2]` as our metric of choice since there is no temptation to use a different average in the denominator.

Does that make sense @YakelynRJ?

Reply:
The interpretation is how much the prediction reduces the MSE compared to predicting no cloud.

Reply:
Yes. @nbren12 and @brianhenn your comments make sense. I will resolve the issues.